### DISCUSSION TITLE
Support for Sucrase

### DISCUSSION THREAD

**Discussion by iagobruno** [vercel/next.js#24822] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.
Not a problem, but in large applications, compile time can take time, even with all the improvements made in Webpack and Next. Time is precious!

## Describe the solution you'd like
[Sucrase](https://github.com/alangpierce/sucrase) is a super fast alternative to Babel (20x faster, according to the maintainer) that transpiles only the codes that are not natively supported by Javascript, such as JSX, Typescript Typing, anyway.
It can drastically decrease Next build time in the development environment. 

Look at the benchmark provided in the package documentation:

```
             Time      Speed
Sucrase      2.928s    225752 lines per second
swc          13.782s   47966 lines per second
TypeScript   39.603s   16693 lines per second
Babel        52.598s   12569 lines per second
```

## Describe alternatives you've considered
I do not know if it is possible to implement this by default in the Next development mode (because the codes have to work in both the browser and nodejs) or if you are interested, but it would be cool to add an example to the /examples folder of how to do the implement it.

**Comment by pablopunk** [comment]
+1

It will also be interesting as it can parse typescript out of the box, no config needed.

**Comment by Timer** [comment]
Our tooling is _highly_ coupled to Babel and we don't see this changing in the future -- especially as we consider compiling `node_modules`. I'm going to close this, but maybe we can revisit later.

**Comment by quantizor** [comment]
@Timer would love to see this reopened and thought about for a future major version of Next. Definitely feel you on Babel buy-in, but the performance is so dramatically different that it could really become a selling point for the framework if the migration took place.

**Comment by janus-reith** [comment]
Apparently the new [Next.js Live](https://nextjs.org/live) feature is using Sucrase, probably to achieve its fast refresh times.
I'm interested to see if this works on regular local dev environments too.

I gave it a very quick try by using `@sucrase/webpack-loader` for .js-files in custom webpack config, and while that worked(Needed to put an React-import to the top of each file containing jsx) I saw no speed improvement, probably because it still passed the regular babel transpilation.

Did anyone else give it a try?

I'm also curious about the new stackblitz webcontainers with next.js support, and wonder how and if these could complement each other or might be related underneath.

**Comment by rolivegab** [comment]
I think that for development purposes, sucrase is a must.
It's super fast
